cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/CassandraDataLayer.java

```diff
@@ -422,7 +424,23 @@ public boolean readIndexOffset()
 
     protected void initInstanceMap()
     {
-        instanceMap = clusterConfig.stream().collect(Collectors.toMap(SidecarInstance::hostname, Function.identity()));
+        if (tokenPartitioner != null)
+        {
+            instanceMap = tokenPartitioner.ring().instances().stream()
+                    .filter(instance -> datacenter == null || datacenter.equals(instance.dataCenter()))
+                    .map(instance -> instance.nodeName()).distinct()
```

Comment:
Can we consider the port as well for the uniqueness check? Just the same as line#353.